### DISCUSSION TITLE
Syncing is failing while awaiting headers

### DISCUSSION THREAD

**Discussion by SoutisV** [argoproj/argo-cd#6426] [Q&A]
Hi, sync is timing out with the usual message: Client.Timeout exceeded while awaiting headers.
On our end we have increased the --repo-server-timeout-seconds and scaled up the argocd-repo-server deployment.

The interesting part is that the git clone from the actual node that argo is running is successful:
![Screen Shot 2021-06-08 at 11 13 49 AM](https://user-images.githubusercontent.com/52630510/121162510-42e15500-c84e-11eb-85e1-afc3c6781ba9.png)

On the other hand, doing the same git clone from argocd-redis pod it is timing out:
![Screen Shot 2021-06-08 at 11 10 57 AM](https://user-images.githubusercontent.com/52630510/121162745-7b812e80-c84e-11eb-9872-651a841370c2.png)
Actually, the git clone from all argocd pods is timing out.

The question is what argo is doing under the hood for syncing?
Any suggestions for troubleshooting and moving forward?

**Comment by jannfis** [comment]
Hey @SoutisV, I believe that the Argo CD pods are not allowed to make outbound connections to the Internet. There may be some Egress network policies for pods in place on your cluster, that prevent Argo CD from accessing the repositories on github.com.

You can double-check whether pods in your namespace are allowed to access GitHub by running a command like the following to start a curl pod in the `argocd` namespace, that will connect to GitHub.com:

```
kubectl run -n argocd -it --rm --image=curlimages/curl curl -- curl -v https://github.com
```
